akka-actor/src/main/scala/akka/dispatch/Future.scala

```diff
@@ -80,13 +82,25 @@ object ExecutionContexts {
    * This is an execution context which runs everything on the calling thread.
    * It is very useful for actions which are known to be non-blocking and
    * non-throwing in order to save a round-trip to the thread pool.
+   *
+   * INTERNAL API
    */
+  // Once Scala 2.12 is no longer supported this can be dropped in favour of directly using [[ExecutionContext.parasitic]]
+  @InternalApi
+  private[akka] val parasitic: ExecutionContext = SameThreadExecutionContext()
```

Comment:
Can we backport this API introduction to 2.5.x so we don't have to rely on deprecated APIs in downstream projects?

Reply:
Yes. I think we can just make the same method defer to the old object to make the backport minimal.

Reply:
See #28715